RecoVertex/V0Producer/src/V0Fitter.cc

```diff
@@ -206,7 +203,8 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
       GlobalPoint vtxPos(theVtx.x(), theVtx.y(), theVtx.z());
 
       // 2D decay significance
-      const SMatrixSym3D totalCov = theBeamSpot->rotatedCovariance3D() + theVtx.covariance();
+      SMatrixSym3D totalCov = theBeamSpot->rotatedCovariance3D() + theVtx.covariance();
+      if (useVertex_) totalCov = referenceVtx.covariance() + theVtx.covariance();
```

Comment:
Hey Vincenzo -

everything I changed only gets implemented if the user specifies
non-standard configuration.
This has to do with either:
1) using a primary vertex instead of a beam spot
2) using z information in calculating variables
by default, the useVertex_ flag will not be set and so this stuff doesn't
get used.

I can change the way the flags / if statements are implemented if you like
- but maybe after the heavy ion run & in whatever the current CMSSW is by
then?...

On Wed, Oct 28, 2015 at 9:06 AM, Vincenzo Innocente <
notifications@github.com> wrote:

> In RecoVertex/V0Producer/src/V0Fitter.cc
> <https://github.com/cms-sw/cmssw/pull/12142#discussion_r43267634>:
>
> > @@ -206,7 +203,8 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
> >        GlobalPoint vtxPos(theVtx.x(), theVtx.y(), theVtx.z());
> >
> >        // 2D decay significance
> > -      const SMatrixSym3D totalCov = theBeamSpot->rotatedCovariance3D() + theVtx.covariance();
> > +      SMatrixSym3D totalCov = theBeamSpot->rotatedCovariance3D() + theVtx.covariance();
> > +      if (useVertex_) totalCov = referenceVtx.covariance() + theVtx.covariance();
>
> same as above why twice?
>
> brw is this fix not changing physics?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12142/files#r43267634>.
>